pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/jdbc/ThinResultSet.java

```diff
@@ -56,10 +60,21 @@ public void close() {
         dataInputStream.close();
       }
       if ( stopped.compareAndSet( false, true ) ) {
-        // TODO issue stop to serviceTrans: http://jira.pentaho.com/browse/BACKLOG-4594
-        ThinDriver.logger.warning( "Need to stop " + thinResultHeader.getServiceTransName() );
+        String id = thinResultHeader.getServiceObjectId();
+        if ( !Const.isEmpty( id ) ) {
+          DataInputStream
+              errorInputStream =
+              client.query( "[ errors " + thinResultHeader.getServiceObjectId() + " ]", 0 );
+          DataInputStream stopInputStream = client.query( "[ stop " + thinResultHeader.getServiceObjectId() + " ]", 0 );
+          Boolean hasErrors = "true".equals( errorInputStream.readUTF() );
+          errorInputStream.close();
+          stopInputStream.close();
```

Comment:
Should these be wrapped in try/finally blocks, incase readUTF fails for some reason?